Explicit backing fields support

dayanruben opened the pull request:
- Handle [Kotlin 2.3.0](https://kotlinlang.org/docs/whatsnew23.html#explicit-backing-fields) explicit backing fields (`field = …` / `field: Type = …`) by visiting `KtBackingField`, emitting _modifiers/name/type/initializer_, and ordering backing fields with accessors by source position so tokenization no longer fails when `-Xexplicit-backing-fields` is enabled.
- Add regression coverage for explicit backing field formatting.

Resolves  #531

hick209 commented on core/src/test/java/com/facebook/ktfmt/format/FormatterTest.kt:
```diff
@@ -4262,6 +4262,21 @@ class FormatterTest {
+          |
+          |  var numbers: List<Int>
+          |    field: MutableList<Int> = mutableListOf()
+          |}
```
Can we also add a test with an accessor as well?
Like `private set` added here.

Also adding modifiers to the backing field would be nice, so we would test that part of the code as well

dayanruben commented:
> Looking great! Thank you so much for helping add support to this feature.
> 
> Please address the comments here and make sure to format the code as well (I think it's `./gradlew ktfmtFormat`).
> 
> Update: I see you are from São Paulo => Mandou bem nesse PR!

Thanks for the review! I’ve already addressed the comments and pushed the changes (including running `./gradlew ktfmtFormat`). Valeu, bora pra cima!

dayanruben commented on core/src/main/java/com/facebook/ktfmt/format/KotlinInputAstVisitor.kt:
```diff
@@ -1431,6 +1453,38 @@ class KotlinInputAstVisitor(
+  private fun emitBackingField(backingField: KtBackingField) {
+    builder.sync(backingField)
+    builder.block(ZERO) {
+      backingField.modifierList?.let { visit(it) }
```
Sorry for the confusion, I rechecked the syntax. You’re right that explicit backing fields don’t accept modifiers.

dayanruben commented:
> Failing unit tests

Perdão. I’ve just pushed the remaining changes, including the changes to remove support for modifiers.

dayanruben commented on core/src/test/java/com/facebook/ktfmt/format/FormatterTest.kt:
```diff
@@ -4278,14 +4278,12 @@ class FormatterTest {
-          |    @Volatile
-          |    field = value
+          |  val count: Int
+          |    field = 0
```
I'm thinking through whether to add a negative test that expects a parsing or formatting error when explicit backing fields are combined with accessors, since the grammar allows them but semantics may forbid it.

dayanruben replied:
Agreed. Do you have a specific accessor case in mind?

hick209 replied:
Nope. Anything that's valid should be fine
